components/cdc/src/service.rs

```diff
@@ -195,132 +232,180 @@ impl Service {
-        });
-
-        let peer = ctx.peer();
-        let scheduler = self.scheduler.clone();
-        ctx.spawn(async move {
-            let res = recv_req.await;
-            // Unregister this downstream only.
+                    Some(ChangeDataRequest_oneof_request::Deregister(_)) => {
+                        Self::handle_deregister(&scheduler, request, conn_id)?
+                    }
+                    _ => unreachable!(),
+                }
+            }
             let deregister = Deregister::Conn(conn_id);
             if let Err(e) = scheduler.schedule(Task::Deregister(deregister)) {
                 error!("cdc deregister failed"; "error" => ?e, "conn_id" => ?conn_id);
             }
-            match res {
-                Ok(()) => {
-                    info!("cdc receive closed"; "downstream" => peer, "conn_id" => ?conn_id);
-                }
-                Err(e) => {
-                    warn!("cdc receive failed"; "error" => ?e, "downstream" => peer, "conn_id" => ?conn_id);
-                }
+            Ok::<(), String>(())
+        };
+
+        let peer = ctx.peer();
+        ctx.spawn(async move {
+            if let Err(e) = recv_req.await {
+                warn!("cdc receive failed"; "error" => ?e, "downstream" => peer, "conn_id" => ?conn_id);
+            } else {
+                info!("cdc receive closed"; "downstream" => peer, "conn_id" => ?conn_id);
             }
         });
 
         let peer = ctx.peer();
-        let scheduler = self.scheduler.clone();
-
         ctx.spawn(async move {
             #[cfg(feature = "failpoints")]
             sleep_before_drain_change_event().await;
-
-            let res = event_drain.forward(&mut sink).await;
-            // Unregister this downstream only.
-            let deregister = Deregister::Conn(conn_id);
-            if let Err(e) = scheduler.schedule(Task::Deregister(deregister)) {
-                error!("cdc deregister failed"; "error" => ?e);
-            }
-            match res {
-                Ok(_s) => {
-                    info!("cdc send closed"; "downstream" => peer, "conn_id" => ?conn_id);
-                    let _ = sink.close().await;
-                }
-                Err(e) => {
-                    warn!("cdc send failed"; "error" => ?e, "downstream" => peer, "conn_id" => ?conn_id);
-                }
+            if let Err(e) = event_drain.forward(&mut sink).await {
+                warn!("cdc send failed"; "error" => ?e, "downstream" => peer, "conn_id" => ?conn_id);
+            } else {
+                info!("cdc send closed"; "downstream" => peer, "conn_id" => ?conn_id);
             }
         });
     }
 }
 
+impl ChangeData for Service {
+    fn event_feed(
+        &mut self,
+        ctx: RpcContext<'_>,
+        stream: RequestStream<ChangeDataRequest>,
+        sink: DuplexSink<ChangeDataEvent>,
+    ) {
+        self.handle_event_feed(ctx, stream, sink, false);
+    }
+
+    fn multiplexing_event_feed(
```

Comment:
How about naming it `event_feed_v2`? So we can add **more** features to the new RPC.